enhancements/helm3/console.md

```diff
@@ -0,0 +1,144 @@
+
+Configuring Helm repository location could be modeled similar to [`OperatorSource`](https://github.com/operator-framework/operator-marketplace/blob/7d230952a1045624b7601b4d6e1d45b3def4cf76/deploy/crds/operators_v1_operatorsource_crd.yaml). 
+Due to future federated usecases ([ODC-2994](https://issues.redhat.com/browse/ODC-2994))), a cluster admin should be able to declare multiple chart repositories.
+
+A cluster-scoped, top-level `HelmRepository` resource)[https://github.com/openshift/api/tree/master/config/v1] should be added:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: HelmRepository
+metadata:
+  name: cluster
+spec:
+  url: http://my.chart-repo.org/stable
+
+  # optional and only needed for UI purposes
+  displayName: myChartRepo
+
+  # optional and only needed for UI purposes
+  description: my private chart repo
+
+  # set to true if need to be disabled temporarly
+  disabled: true
+```
+
+The console operator would watch for changes on it and reconfigure the chart repository proxy. The console backend already implements a few helm endpoints (including the chart proxy), but the future might require to extract them into a separate service (e.g. to make openshift more modular). Hence, it would be good to define the configuration as the top-level resource detaching API from the implementation. Cluster admin would easily discover the new resource and manage it either via CLI or through UI. 
+
+![Helm Cluster Configuration](assets/openshift-administration-cluster-settings.png)
+
+#### Alternatives
+
+#### 1. The configuration could be embedded into cluster-wide [`Console` config](https://github.com/openshift/api/blob/master/config/v1/types_console.go#L26)
+
+Admins wouldn't be able to intuitively discover the operator config as a way to configure the Helm repository URLs. It becomes closely coupled with the console. Extracting Helm endpoints into a separate service would require moving the config as well.
+
+#### 2. The configuration could be embedded into [`Console` operator config](https://github.com/openshift/api/blob/master/operator/v1/types_console.go#L26)
+
+Conceptually, the Helm repository URL isn't really an operator configuration, hence this doesn't feel like the right place.
+This approach would have similar issues with the previous alternative - admins wouldn't be able to intuitively discover the operator config as a way to configure the Helm repository URLs.
+
+#### 3. OLM operator for Helm Configuration. 
+
+Note, the helm charts' repository configuration today exists as a console configuration, which enables Console to proxy to the Helm chart repository URL. Moving it out of Console is outside the scope of this section. 
+
+   * The default helm chart repository URL remains unchanged in the Console configuration.
+   * Admin installs an OLM operator which only provides a `HelmConfig` cluster-scoped CRD
+   * Admin creates a cluster-scoped CR. Note, this isn't very intuitive for the Admin.
```

Comment:
this looks spurious.  It's equivalent to what you're trying to do here.

Reply:
Here the admin configures a cluster-wide helm chart repository by creating a CR with the details.

Reply:
i dont understand how this is not intuitive.  this is equivalent to config.openshift.io use case.

Reply:
@derekwaynecarr , an admin need to understand that changing the default chart repo location requires installing an OLM operator.